dynamic/clientset/clientset.go

```diff
@@ -150,3 +151,27 @@ func (rc *ResourceClient) AtomicUpdate(orig *unstructured.Unstructured, update f
 	})
 	return result, err
 }
+
+// AddFinalizer adds the given finalizer to the list, if it isn't there already.
+func (rc *ResourceClient) AddFinalizer(orig *unstructured.Unstructured, name string) (*unstructured.Unstructured, error) {
+	return rc.AtomicUpdate(orig, func(obj *unstructured.Unstructured) bool {
+		if dynamicobject.HasFinalizer(obj, name) {
```

Comment:
This `if` clause can be skipped if it turns out that it is fine for `AddFinalizer` to return a `bool` value (refer below).

Reply:
I thought about that, but ultimately I felt like the call sites for `AddFinalizer` will be more readable if they explicitly call `HasFinalizer` when it matters, as is done here. It's an extra loop pass over the list, but I have no data pointing to this as a performance concern so I chose readability over brevity or premature optimization.

Reply:
noted!